php-checks/src/main/java/org/sonar/php/checks/EvalUseCheck.java

```diff
@@ -19,34 +19,37 @@
  */
 package org.sonar.php.checks;
 
-import com.sonar.sslr.api.AstNode;
 import org.sonar.api.server.rule.RulesDefinition;
 import org.sonar.check.BelongsToProfile;
 import org.sonar.check.Priority;
 import org.sonar.check.Rule;
 import org.sonar.php.api.PHPKeyword;
+import org.sonar.php.checks.utils.FunctionUsageCheck;
+import org.sonar.plugins.php.api.tree.expression.FunctionCallTree;
 import org.sonar.squidbridge.annotations.SqaleConstantRemediation;
 import org.sonar.squidbridge.annotations.SqaleSubCharacteristic;
-import org.sonar.squidbridge.checks.SquidCheck;
-import org.sonar.sslr.parser.LexerlessGrammar;
 
 @Rule(
-  key = "S1523",
+  key = EvalUseCheck.KEY,
   name = "Code should not be dynamically injected and executed to prevent Eval Injection vulnerability",
   priority = Priority.CRITICAL,
   tags = {Tags.SECURITY, Tags.CWE, Tags.OWASP_A3})
 @BelongsToProfile(title = CheckList.SONAR_WAY_PROFILE, priority = Priority.MAJOR)
 @SqaleSubCharacteristic(RulesDefinition.SubCharacteristics.INPUT_VALIDATION_AND_REPRESENTATION)
 @SqaleConstantRemediation("30min")
-public class EvalUseCheck extends SquidCheck<LexerlessGrammar> {
+public class EvalUseCheck extends FunctionUsageCheck {
+
+  public static final String KEY = "S1523";
+  private static final String MESSAGE = "Remove this use of the \"eval\" function.";
 
   @Override
-  public void init() {
-    subscribeTo(PHPKeyword.EVAL);
+  protected String functionName() {
+    return PHPKeyword.EVAL.getValue();
   }
 
   @Override
-  public void visitNode(AstNode astNode) {
-    getContext().createLineViolation(this, "Remove this use of the \"eval\" function.", astNode);
+  protected void createIssue(FunctionCallTree tree) {
+    context().newIssue(this.KEY, MESSAGE).tree(tree.callee());
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Change this instance-reference to a static reference. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2209)